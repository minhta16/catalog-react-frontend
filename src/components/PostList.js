import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import PostItem from './PostItem';

export function PostList(props) {
  const { posts } = props;
  return (
    <div style={{ margin: '.5rem' }}>
      <Typography variant="h4" className="center-margin">
        Your Posts
      </Typography>
      {posts.reverse().map((post) => (
        <PostItem key={post.name} post={post} />
      ))}
    </div>
  );
}

PostList.propTypes = {
  posts: PropTypes.array.isRequired,
};

export default PostList;
