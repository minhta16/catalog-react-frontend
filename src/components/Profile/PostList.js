import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Typography } from '@material-ui/core';
import { openSnackbar as openSnackbarRedux } from 'actions/misc';
import PostItem from './PostItem';

export class PostList extends Component {
  state = {};

  handleOpenSnackbar = () => {
    const { openSnackbar } = this.props;
    openSnackbar('Post created!');
  };

  render() {
    const { posts } = this.props;
    return (
      <div style={{ margin: '.5rem' }}>
        <Typography variant="h4" className="center-margin">
          Your Posts
        </Typography>
        {posts.length > 0 ? (
          posts.map((post) => <PostItem key={post.id} post={post} />)
        ) : (
          <Typography className="center-margin" variant="body1">
            You have no post. Go ahead and create some. Our community would appreciate your posts!
          </Typography>
        )}
      </div>
    );
  }
}

PostList.propTypes = {
  posts: PropTypes.array.isRequired,
  openSnackbar: PropTypes.func.isRequired,
};

export default connect(
  null,
  { openSnackbar: openSnackbarRedux },
)(PostList);
