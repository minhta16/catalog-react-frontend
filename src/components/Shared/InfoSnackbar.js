import React from 'react';
import PropTypes from 'prop-types';

import { Snackbar, IconButton } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

export function InfoSnackbar(props) {
  const { open, onClose, message } = props;
  return (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
      open={open}
      autoHideDuration={5000}
      onClose={onClose}
      message={message}
      action={[
        <IconButton key="close" aria-label="Close" color="inherit" onClick={onClose}>
          <CloseIcon />
        </IconButton>,
      ]}
    />
  );
}

InfoSnackbar.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func,
  message: PropTypes.string,
};

InfoSnackbar.defaultProps = {
  open: false,
  onClose: () => {},
  message: '',
};

export default InfoSnackbar;
