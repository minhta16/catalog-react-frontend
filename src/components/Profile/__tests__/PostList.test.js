/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import InfoSnackbar from 'components/Shared/InfoSnackbar';
import { PostList } from '../PostList';

describe('components/Profile/PostList', () => {
  let wrapper;
  let props;
  // let warning;

  // const update = () => {
  //   wrapper.update();
  //   warning = wrapper.find(Warning);
  // };

  const setup = () => {
    wrapper = shallow(<PostList {...props} />);
  };

  beforeEach(() => {
    props = {
      posts: [
        {
          name: 'bro',
          description: 'bro',
          category_id: '1',
          created: '2019',
          id: '1',
        },
      ],
    };
  });

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call closeSnackbar correctly', () => {
    setup();
    const snackbar = wrapper.find(InfoSnackbar);
    snackbar.simulate('close');
    expect(wrapper.state().openSnackbar).toBe(false);
  });
});
