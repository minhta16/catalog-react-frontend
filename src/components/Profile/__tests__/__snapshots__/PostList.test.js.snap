// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Profile/PostList should render correctly 1`] = `
<div
  style={
    Object {
      "margin": ".5rem",
    }
  }
>
  <WithStyles(ForwardRef(Typography))
    className="center-margin"
    variant="h4"
  >
    Your Posts
  </WithStyles(ForwardRef(Typography))>
  <PostItem
    key="bro"
    post={
      Object {
        "category_id": "1",
        "created": "2019",
        "description": "bro",
        "name": "bro",
      }
    }
    subtitleLength={100}
  />
</div>
`;
